include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -7377,8 +7377,6 @@ FEEvaluation<dim,
   Assert(this->mapped_geometry == nullptr,
          ExcMessage("FEEvaluation was initialized without a matrix-free object."
                     " Integer indexing is not possible"));
-  if (this->mapped_geometry != nullptr)
```

Comment:
Why is this no longer needed?

Reply:
We do the assert immediately above, which indicates we do not actually want to support this case. Now the release mode would segfault somewhere, but that is acceptable and in my opinion better than trying to fix up for this case.